hibernate-reactive-core/src/main/java/org/hibernate/reactive/event/impl/AbstractReactiveSaveEventListener.java

```diff
@@ -182,20 +182,21 @@ private CompletionStage<Object> generateId(
 			EntityPersister persister) {
 		return generator
 				.generate( (ReactiveConnectionSupplier) source, entity )
-				.thenApply( id -> castToIdentifierType( id, persister ) )
-				.thenCompose( generatedId -> {
-					if ( generatedId == null ) {
-						return failedFuture( new IdentifierGenerationException( "null id generated for: " + entity.getClass() ) );
-					}
-					if ( LOG.isDebugEnabled() ) {
-						LOG.debugf(
-								"Generated identifier: %s, using strategy: %s",
-								persister.getIdentifierType().toLoggableString( generatedId, source.getFactory() ),
-								generator.getClass().getName()
-						);
-					}
-					return completedFuture( generatedId );
-				} );
+				.thenApply( id -> {
+								final Object generatedId = castToIdentifierType( id, persister );
+								if ( generatedId == null ) {
+									return failedFuture( new IdentifierGenerationException( "null id generated for: " + entity.getClass() ) );
```

Comment:
Now that this is a `.thenApply`, you have to throw the exception like in the original code. Otherwise, the error will be ignored

Reply:
Same here: https://github.com/hibernate/hibernate-reactive/pull/2082